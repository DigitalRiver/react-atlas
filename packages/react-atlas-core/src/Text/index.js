import { default as TextCore } from "./Text";
export { TextCore };