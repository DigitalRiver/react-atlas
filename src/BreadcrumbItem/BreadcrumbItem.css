.breadcrumbItem {
    display: inline-flex;
    align-items: center;
}

.breadcrumbItem + .breadcrumbItem:before {
    content: "/ ";
    padding: 0 5px;
    color: #ccc;
}

.breadcrumbItem.active {
    color: #777;
}