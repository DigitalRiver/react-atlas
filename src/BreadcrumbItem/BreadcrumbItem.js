import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./BreadcrumbItem.css";

export class BreadcrumbItem extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const { className, style, active, children } = this.props;
    const styleName = cx("breadcrumbItem", { active });
    return (
      <li styleName={styleName} className={cx(className)} style={style}>
        {children}
      </li>
    );
  }
}

BreadcrumbItem.propTypes = {
  /** An object, array, or string of CSS classes to apply to BreadcrumbItem.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object,
  /**
   * Text, any HTML element, or React Component.
   */
  "children": PropTypes.node,
  /**
   * Render active item style
   */
  "active": PropTypes.bool
};

BreadcrumbItem.defaultProps = {
  "className": "",
  "active": false
};

export default CSSModules(BreadcrumbItem, styles, { "allowMultiple": true });
