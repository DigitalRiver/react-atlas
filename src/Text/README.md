###### Default Text component:

    <Text>Hello World</Text>

###### Nested Text components:

    <Text>Hello <Text style={{color: "red"}}>World</Text></Text>

###### Nested view components:

    <Text>There is a Button <Button>Hello</Button> inside my Text.</Text>

###### Default Text for body copy:

    <Text body>Hello World</Text>

###### Input label:

    <div>
	  <Text as="label" htmlFor="inputId">Input Label</Text>
	  <TextField id="inputId" />
	</div>

###### Headers and Sub-Headers:

    <div>
	    <Text as="h1">h1 Heading <Text as="small">Secondary text</Text></Text>
	    <Text as="h2">h2 Heading <Text as="small">Secondary text</Text></Text>
	    <Text as="h3">h3 Heading <Text as="small">Secondary text</Text></Text>
	    <Text as="h4">h4 Heading <Text as="small">Secondary text</Text></Text>
	    <Text as="h5">h5 Heading <Text as="small">Secondary text</Text></Text>
	    <Text as="h6">h6 Heading <Text as="small">Secondary text</Text></Text>
	</div>

###### Paragraph Text:

    <div>
    	<Text as="p">Parapraph One</Text>
    	<Text as="p">Paragraph Two</Text>
    </div>

###### Bold text:

    <Text as="b">Hello World</Text>

###### Important text:

    <Text as="strong">Hello World</Text>

###### Italic text:

    <Text as="i">Hello World</Text>

###### Emphasized text:

    <Text as="em">Hello World</Text>

###### Deleted text:

    <Text as="del">Hello World</Text>

###### Inserted text:

    <Text as="ins">Hello World</Text>

###### Marked text:

    <Text as="mark">Hello World</Text>

###### Small text (must be used within another Text component):

    <Text>Hello <Text as="small">World</Text></Text>

###### Subscript text:

    <Text>Hello <Text as="sub">World</Text></Text>

###### Superscript text:

    <Text>Hello <Text as="sup">World</Text></Text>

###### Code text:

    <Text>Sample Code: <Text as="code">Hello World</Text></Text>

###### Blockquote text:

    <Text as="blockquote">Hello World</Text>

###### Link text:

    <Text as="a" href="#">Hello World</Text>

