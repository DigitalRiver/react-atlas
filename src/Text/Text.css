@import '../styles/variables.css';

.label {
	composes: header-font from '../styles.css';
}

.defaultFont {
	composes: default-font from '../styles.css';
}

.bodyFont {
	composes: body-font from '../styles.css';
}

.header {
	composes: header-font from '../styles.css';
	font-weight: 500;
	line-height: 1;
	display: block;
}

.h1 {
	composes: header;
	font-size: 52px;
	margin: 0 0 13.5px;
}

.h2 {
	composes: header;
	font-size: 42px;
	margin: 0 0 13.5px;
}

.h3 {
	composes: header;
	font-size: 35px;
	margin: 0 0 13.5px;
}

.h4 {
	composes: header;
	font-size: 26px;
	margin: 0 0 6.75px;
}

.h5 {
	composes: header;
	font-size: 21px;
	margin: 0 0 6.75px;
}

.h6 {
	composes: header;
	font-size: 19px;
	margin: 0 0 6.75px;
}

.small {
	composes: dark-gray from '../styles.css';
	font-size:65%;
}

.link {
	composes: sky-blue from '../styles.css';
	text-decoration: none;
}

.link:hover {
  color: var(--secondary);
}

.code {
	composes: red from '../styles.css';
    composes: bg-silver from '../styles.css';
    composes: pad-h-0-25 from '../styles.css';
}

.blockquote {
	padding: var(--space-1-1) var(--space-2);
	composes: pad-v-1 from '../styles.css';
	composes: pad-h-2 from '../styles.css';
    margin: 0 0 var(--space-2);
    font-size: var(--header-font-size);
    border-left: 5px solid var(--silver);
}