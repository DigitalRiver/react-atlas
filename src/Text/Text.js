import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./Text.css";

export class Text extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      as,
      body,
      children,
      className,
      htmlFor,
      id,
      style,
      ...other
    } = this.props;

    const textStyles = cx({
      "blockquote": as === "blockquote",
      "bodyFont": body,
      "code": as === "code",
      "defaultFont": !body && as !== "label",
      "h1": as === "h1",
      "h2": as === "h2",
      "h3": as === "h3",
      "h4": as === "h4",
      "h5": as === "h5",
      "h6": as === "h6",
      "label": as === "label",
      "link": as === "a",
      "small": as === "small"
    });

    const TextElement = `${this.props.as}`;

    let element = 
      <TextElement
        id={id}
        htmlFor={htmlFor}
        style={style}
        className={className}
        styleName={textStyles}
        {...other}
      >
        {children}
      </TextElement>
    ;

    return element;
  }
}

Text.propTypes = {
  /** Defines what HTML element to render the Text component as. Possibilities include, but are not limited to b, body, del, em, h1, h2, h3, h4, h5, h6, i, ins, label, mark, p, small, strong, sub, and sup. */
  "as": PropTypes.string,
  /** Allows theme to differentiate font-family between default font and body copy in variables.css */
  "body": PropTypes.bool,
  /**
   * Text, any HTML element, or React Component.
   */
  "children": PropTypes.node,
  /** An Object, array, or string of CSS classes to apply to Input. */
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /** Adds a "for" attribute to your element. Intended to be used when setting the as property to "label". */
  "htmlFor": PropTypes.string,
  /**
   * Defines an id for the input.
   * @examples '<Text as="text" name="test"/>'
   */
  "id": PropTypes.string,
  /** Pass inline styling here. */
  "style": PropTypes.object
};

Text.defaultProps = {
  "as": "span"
};

export default CSSModules(Text, styles, { "allowMultiple": true });
