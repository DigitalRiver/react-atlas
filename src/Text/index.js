import Text from "./Text.js";
export { Text };
