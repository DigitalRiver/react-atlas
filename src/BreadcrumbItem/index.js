import BreadcrumbItem from "./BreadcrumbItem.js";
export { BreadcrumbItem };
