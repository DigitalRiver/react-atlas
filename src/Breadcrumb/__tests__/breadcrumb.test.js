import React from "react";
import { mount } from "enzyme";
import { Breadcrumb } from "../index";
import { BreadcrumbItem } from "../../BreadcrumbItem/index";
import { Text } from "../../Text/index";

import renderer from "react-test-renderer";

describe("Test Breadcrumb component", () => {
  it("Test correct render", function() {
    const tree = renderer
      .create(
        <Breadcrumb>
          <BreadcrumbItem>
            <Text as="a" href="#">
              Atlas
            </Text>
          </BreadcrumbItem>
          <BreadcrumbItem>
            <Text as="a" href="#breadcrumb">
              Breadcrumb
            </Text>
          </BreadcrumbItem>
          <BreadcrumbItem active>
            <Text>Breadcrumb Example</Text>
          </BreadcrumbItem>
        </Breadcrumb>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Testing Breadcrumb component: Basic Tests", () => {
  const brdcrmb = mount(
    <Breadcrumb>
      <BreadcrumbItem>
        <Text as="a" href="#">
          Atlas
        </Text>
      </BreadcrumbItem>
      <BreadcrumbItem>
        <Text as="a" href="#breadcrumb">
          Breadcrumb
        </Text>
      </BreadcrumbItem>
      <BreadcrumbItem active>
        <Text>Breadcrumb Example</Text>
      </BreadcrumbItem>
    </Breadcrumb>
  );

  it("Basic Breadcrumb", () => {
    expect(brdcrmb.exists(<Breadcrumb />)).toBe(true);
  });

  it("Contains No Child", () => {
    const breadcrumb = mount(<Breadcrumb />);

    expect(breadcrumb.props().children).toBeUndefined();
  });

  it("Contains Child", () => {
    expect(brdcrmb.props()).toHaveProperty("children");
  });

  it("Basic Breadcrumb with BreadcrumbItem child", () => {
    expect(
      brdcrmb.contains(
        <BreadcrumbItem>
          <Text as="a" href="#">
            Atlas
          </Text>
        </BreadcrumbItem>
      )
    ).toBeTruthy();
  });
});
