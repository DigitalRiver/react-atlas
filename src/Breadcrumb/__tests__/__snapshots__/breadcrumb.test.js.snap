// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Breadcrumb component Test correct render 1`] = `
<ol
  aria-label="breadcrumbs"
  className="breadcrumb"
  role="navigation"
  style={undefined}
>
  <li
    className="breadcrumbItem"
    style={undefined}
  >
    <a
      className="defaultFont link"
      href="#"
      htmlFor={undefined}
      id={undefined}
      style={undefined}
    >
      Atlas
    </a>
  </li>
  <li
    className="breadcrumbItem"
    style={undefined}
  >
    <a
      className="defaultFont link"
      href="#breadcrumb"
      htmlFor={undefined}
      id={undefined}
      style={undefined}
    >
      Breadcrumb
    </a>
  </li>
  <li
    className="breadcrumbItem active"
    style={undefined}
  >
    <span
      className="defaultFont"
      htmlFor={undefined}
      id={undefined}
      style={undefined}
    >
      Breadcrumb Example
    </span>
  </li>
</ol>
`;
